dwave_networkx/algorithms/canonicalization.py

```diff
@@ -0,0 +1,161 @@
+# Copyright 2018 D-Wave Systems Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import division
+
+
+import math
+
+from dwave_networkx.generators.chimera import chimera_coordinates
+
+__all__ = ['canonical_chimera_labeling']
+
+
+def canonical_chimera_labeling(G, t=None):
+    """Returns a mapping from the labels of G to chimera-indexed labeling.
+
+    Parameters
+    ----------
+    G : NetworkX graph
+        A Chimera-structured graph.
+    t : int (optional, default 4)
+        Size of the shore within each Chimera tile.
+
+    Returns
+    -------
+    chimera_indices: dict
+        A mapping from the current labels to a 4-tuple of Chimera indices.
+
+    """
+    adj = G.adj
+
+    if t is None:
+        if hasattr(G, 'edges'):
+            num_edges = len(G.edges)
+        else:
+            num_edges = len(G.quadratic)
```

Comment:
Potential five lines to one:
```
num_edges = len(G.edges) if hasattr(G, 'edges') else len(G.quadratic)
```

Reply:
I think the 5 lines is clearer, but can be convinced otherwise.

Reply:
Haha, if you want even shorter (and unclearer):
```
num_edges = len(getattr(G, 'edges', None) or G.quadratic))
```
(assuming G.edges is never 0)